apps/root/src/pages/transfer/transfer-container/components/token-selector/index.tsx

```diff
@@ -0,0 +1,113 @@
+import styled from 'styled-components';
+import React from 'react';
+import { Token } from '@types';
+import TokenPickerModal from '@common/components/token-picker-modal';
+import TokenPickerWithAmount from '@common/components/token-picker-with-amount';
+import useActiveWallet from '@hooks/useActiveWallet';
+import useReplaceHistory from '@hooks/useReplaceHistory';
+import { addCustomToken } from '@state/token-lists/actions';
+import useUsdPrice from '@hooks/useUsdPrice';
+import { parseUnits } from '@ethersproject/units';
+import useBalance from '@hooks/useBalance';
+import { useAppDispatch } from '@state/hooks';
+import { useTransferState } from '@state/transfer/hooks';
+import { setAmount, setToken } from '@state/transfer/actions';
+import useToken from '@hooks/useToken';
+import { FormattedMessage } from 'react-intl';
+import useSelectedNetwork from '@hooks/useSelectedNetwork';
+
+interface TokenSelectorProps {
+  tokenParamAddress?: string;
+}
+
+const StyledContentContainer = styled.div`
+  position: relative;
+  padding: 16px;
+  border-radius: 8px;
+  gap: 16px;
+  display: flex;
+  flex-direction: column;
+`;
+
+const TokenSelector = ({ tokenParamAddress }: TokenSelectorProps) => {
+  const dispatch = useAppDispatch();
+  const replaceHistory = useReplaceHistory();
+  const selectedNetwork = useSelectedNetwork();
+  const activeWallet = useActiveWallet();
+  const { token: selectedToken, amount, recipient } = useTransferState();
+  const [balance] = useBalance(selectedToken, activeWallet?.address);
+  const tokenParam = useToken(tokenParamAddress, undefined, true);
+
+  const [fetchedTokenPrice, loadingTokenPrice] = useUsdPrice(
+    selectedToken,
+    parseUnits(amount || '0', selectedToken?.decimals)
+  );
+  const [shouldShowPicker, setShouldShowPicker] = React.useState(false);
+
+  const cantFund = !!selectedToken && !!amount && !!balance && parseUnits(amount, selectedToken.decimals).gt(balance);
+
+  React.useEffect(() => {
+    if (tokenParam) {
```

Comment:
why does this need to change when the chainId changes?

Reply:
I used the `chainId` to update the component when the default network was overwritten. Thanks to this comment, I was able to find a better solution by using `tokenParam` as a dependency and only updating Redux when there is no `selectedToken`.